### PR TITLE
fix: check resourceowner not empty string

### PR DESCRIPTION
**Pull request by livio-a**

### CODE REVIEW

File: internal/eventstore/eventstore_test.go [hunk 1]
```diff
@@ -446,6 +447,143 @@ func TestEventstore_aggregatesToEvents(t *testing.T) {
...
+				wantErr: true,
+			},
+		},
+		{
+			name: "no resourceowner",
+			args: args{
+				events: []Command{
+					&testEvent{
+						BaseEvent: *NewBaseEventForPush(
+							service.WithService(authz.NewMockContext("resourceOwner", "editorUser"), "editorService"),
```
Comment (adlerhurst) [inline internal/eventstore/eventstore_test.go:556]: Empty resource owner in mock ctx

File: internal/eventstore/repository/event.go [hunk 2]
```diff
@@ -54,7 +55,7 @@ type Event struct {
 	//ResourceOwner is the organisation which owns this aggregate
 	// an aggregate can only be managed by one organisation
 	// use the ID of the org
-	ResourceOwner string
+	ResourceOwner sql.NullString
```
Comment (adlerhurst) [inline internal/eventstore/repository/event.go:58]: Shouldnt a nullstring be provided as pointer to de sql psckage?

File: internal/command/org_member.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func (c *Commands) addOrgMember(ctx context.Context, orgAgg *eventstore.Aggregat
 	if !member.IsValid() {
 		return nil, caos_errs.ThrowInvalidArgument(nil, "Org-W8m4l", "Errors.Org.MemberInvalid")
 	}
-	if len(domain.CheckForInvalidRoles(member.Roles, domain.OrgRolePrefix, c.zitadelRoles)) > 0 {
+	if len(domain.CheckForInvalidRoles(member.Roles, domain.OrgRolePrefix, c.zitadelRoles)) > 0 && len(domain.CheckForInvalidRoles(member.Roles, domain.RoleSelfManagementGlobal, c.zitadelRoles)) > 0 {
```
Comment (livio-a) [inline internal/command/org_member.go:45]: see https://github.com/caos/zitadel/pull/2818#discussion_r778594584

File: internal/eventstore/repository/event.go (hunk 2 above)
Comment (livio-a) [reply internal/eventstore/repository/event.go:58]: why? the nullstring handles the tristate itself and the only reason for a pointer is the scan method

**Review by adlerhurst — DISMISSED** [review]

**Review by fgerschwiler — APPROVED** [review]

**Pull request merged**